src/producer.rs

```diff
@@ -242,15 +237,34 @@ pub fn to_substrait_rel(
                 }
             }
             // map the left and right columns to binary expressions in the form `l = r`
-            let join_expression: Vec<Expr> = join
-                .on
-                .iter()
-                .map(|(l, r)| Expr::Column(l.clone()).eq(Expr::Column(r.clone())))
-                .collect();
+            let join_expression: Vec<Expr>;
```

Comment:
I prefer not to declare first and initialize later. Maybe we can use this instead?
```rust
let join_expression = if join.null_equals_null {...} else {...};
```